scripts/sync-csi-repos-with-upstream.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+source_target_repos=(
+    "external-snapshotter:csi-snapshotter"
+    "external-resizer:csi-resizer"
+    "external-provisioner:csi-provisioner"
+    "external-attacher:csi-attacher"
+    "node-driver-registrar:csi-node-driver-registrar"
+    "livenessprobe:livenessprobe"
+)
+
+for source_target_repo in "${source_target_repos[@]}"; do
+    source_repo=$(echo "$source_target_repo" | cut -d ":" -f 1)
+    target_repo=$(echo "$source_target_repo" | cut -d ":" -f 2)
+
+    echo "Syncing $target_repo"
+
+    SOURCE_REPO_PATH="https://github.com/kubernetes-csi/${source_repo}.git"
+    TARGET_REPO_PATH="https://github.com/longhorn/${target_repo}.git"
+
+    git clone ${TARGET_REPO_PATH}
+    pushd ${target_repo}
+
+    # Sync branches
+    git remote add src_remote ${SOURCE_REPO_PATH}
+    git fetch src_remote --quiet
+    git push origin "refs/remotes/src_remote/*:refs/heads/*" --force
+
+    # Sync tags
+    git tag -d $(git tag -l)
```

Comment:
Will this delete our downstream tags? Do we need to delete all tags first?

Reply:
Just make sure the downstream tags need to be kept after syncing from the source repo.

Reply:
The script won't delete any tags in the target repo (our forked repo).